app/app/static/custom/ourscript.js

```diff
@@ -4,85 +4,202 @@
 *
 */
 
-var inSessionTable = $('#inSession tbody');
-var timers = [];
-var timerIntervals = [];
 
+const timers = [];
+const timerIntervals = [];
+
+const queueList = {
+	"STUDYSmarter":[],
+	"Librarian":[],
+	"In Session": []
+}
+
+const addToQueueList = async (data) => {
+	// This function adds the data (object - of the student details) to the specified queueList
+	// And rerenders the table
+
+	try{
+		const response = await fetch("add_entry", {
+			method: "POST",
+			body: JSON.stringify(data),
+			headers: {
+				'Content-Type': 'application/json'
+			},
+		})
+		const dataResponse = await response.json()
+		queueList[data.queue].push(dataResponse)
+		
+		timers[dataResponse.id] = 0;
+		timerIntervals[dataResponse.id] = setInterval(setTime, 1000, dataResponse.id);
+		rerenderTables()
+	}
+	catch(error){
+		// There's an error
+		console.log(error)
+	}
+}
+
+const moveToSessionList = async(data) => {
+	// This function move the data (object - of the student details) to the in Session table
+	// And rerenders the table
+	try{
+		const response = await fetch(`/update_entry/${data.id}`, {
+			method: "POST",
+			body: JSON.stringify(data),
+			headers: {
+				'Content-Type': 'application/json'
+			},
+		})
+		const dataResponse = await response.json()
+		const index = queueList[dataResponse.queue].findIndex((element) => element.id == dataResponse.id)
+		// pop the data from waiting queue
+		queueList[dataResponse.queue].splice(index, 1)
+		queueList['In Session'].push(dataResponse)
+
+		clearInterval(timerIntervals[dataResponse.id]);
+		timers[dataResponse.id] = 0;
+		timerIntervals[dataResponse.id] = setInterval(setTime, 1000, dataResponse.id);
+		rerenderTables()
+	}
+	catch(error){
+		// There's an error
+		console.log(error)
+	}
+}
+
+const terminateSession = async(data) => {
+	// This function update the status data 
+	// And rerenders the table
+	try{
+		const response = await fetch("/update_entry/" + data.id, {
+			method: "POST",
+			body: JSON.stringify(data),
+			headers: {
+				'Content-Type': 'application/json'
+			},
+		})
+		const dataResponse = await response.json()
+		if (dataResponse.status == "Ended"){
+			const index = queueList[dataResponse.queue].findIndex((element) => element.id == dataResponse.id)
+			// pop the data from current queue
+			queueList[dataResponse.queue].splice(index, 1)
```

Comment:
Do you not need to rerender table after this operation?

Reply:
I put rerender after the if else command
https://github.com/UWA-CITS3200-18-2021/ReSQ/blob/2770643dd6f69f02fe7122243d00a06c474d6e3c/app/app/static/custom/ourscript.js#L87-L94